docs/library/uasyncio.rst

```diff
@@ -136,6 +139,29 @@ class Event
 
     This is a coroutine.
 
+class ThreadSafeFlag
+--------------------
+
+.. class:: ThreadSafeFlag()
+
+    Create a new flag which can be used to synchronise a task with code running
+    outside the asyncio loop, such as other threads, IRQs, or scheduler
+    callbacks.  Flags start in the cleared state.
+
+.. method:: ThreadSafeFlag.set()
+
+    Set the flag.  If there is a task waiting on the event, it will be scheduled
+    to run.
+
+.. method:: ThreadSafeFlag.wait()
+
+    Wait for the flag to be set.  If the flag is already set then it returns
+    immediately.
```

Comment:
that doesn't match the implementation (returning immediately if set)... maybe the implementation should be changed so it does return immediately if `self._flag` is set... that would make things more responsive at the risk of "starving" other things (but since Event.wait already behaves this way we know it's probably ok to return immediately)

Reply:
Yeah I agree. I've updated the code to return immediately and added that to the test.